### ISSUE TITLE
allow immutable state variables to be named in CAPS_SNAKE_CASE

### ISSUE THREAD

**Issue by quezak** [protofire/solhint#243]
This is more of a question than a bug report -- IMO [immutable variables](https://solidity.readthedocs.io/en/v0.7.0/contracts.html#constant-and-immutable-state-variables) should be allowed to be treated as constants, since they in fact cannot be changed.

Currently the code below returns a `warning  Variable name must be in mixedCase` when linted with the recommended ruleset.

```solidity
contract Test {
  uint16 public immutable TEST_MODE;
  constructor(uint16 _testMode) { TEST_MODE = _testMode; }
}
```

**Comment by fvictorio** [comment]
I think we need to rethink the whole approach to naming, since it's hard to make some calls. For example, the linked docs use mixedCase for immutable variables, but I agree that it makes sense to use SNAKE_CASE.

The quick and backward compatible fix here is to add an option to the `var-name-mixedcase`. Something like `{ ignoreImmutable: true }`. The problem is, if you do that you would also need to add a mirror option to `const-name-snakecase`, something like `{ allowImmutable: true }`. What do you think?

**Comment by quezak** [comment]
It's hard to think of a good single setting here, since it really concerns two rules. The simplest way would be to have some "general config" option `treat-immutable-vars-as-constants`, but that's kinda bad design since it's not really a general option, just a two-rule option...
Maybe it will be enough to add a `treat-immutable-var-as-constant: true` option to both these rules? You'd have to turn on both to have immutable variables linted in `CAPS_SNAKE_CASE`, and without them it would lint them in `mixedCase`.

**Comment by fvictorio** [comment]
Yeah, I'd be fine with that. Especially because having such a long and ugly option would motivate me to totally redo the naming rules in the next major :smile:

**Comment by benesjan** [comment]
I don't know what is the state of this issue now but having support for this would be great 👍

**Comment by maxholloway** [comment]
Plus one, would love to see this!

**Comment by smartcontracts** [comment]
+1 on this, would be great, Optimism prefers `SCREAMING_SNAKE_CASE` for immutables

**Comment by hktitof** [comment]
+1 here too guyes

**Comment by yuradmt** [comment]
agree, let's add the snake case

**Comment by clicman** [comment]
Are any updates?

**Comment by SvenMeyer** [comment]
yes,please !

**Comment by dbale-altoros** [comment]
see this PR
https://github.com/protofire/solhint/pull/458
will add new rule to treat immutable as constants or mixed case as needed
